### PR TITLE
storage_service: make all Raft-based operations abortable

### PR DESCRIPTION
**Pull request by patjed41**
During a shutdown, we call `storage_service::stop_transport` first.
We may try to apply a Raft command after that, or still be in the
the process of applying a command. In such a case, the shutdown
process will hang because Raft retries replicating a command until
it succeeds even in the case of a network error. It will stop when
a corresponding abort source is set. However, if we pass `nullptr`
to a function like `add_entry`, it won't stop. The shutdown
process will hang forever.

We fix all places that incorrectly pass `nullptr`. These shutdown
hangs are not only theoretical. The incorrect `add_entry` call in
`update_topology_state` caused scylladb/scylladb#16435.

Additionally, we remove the default `nullptr` values in all member
functions of `server` and `raft_group0_client` to avoid similar bugs
in the future.

Fixes scylladb/scylladb#16435

### CODE REVIEW

**Comment by nyh** [comment]
@kbr-scylla please review (or ask someone else to review). This issue is breaking CI too frequently.

**Comment by nyh** [comment]
There are several duplicate issues for https://github.com/scylladb/scylladb/issues/16435. Let's either close the duplicates now, or put a list of "Fixes" lines on this PR to close all of them.

File: service/broadcast_tables/experimental/lang.cc [hunk 1]
```diff
@@ -30,7 +30,7 @@ bool is_broadcast_table_statement(const sstring& keyspace, const sstring& column
 future<query_result> execute(service::raft_group0_client& group0_client, const query& query) {
     auto group0_cmd = group0_client.prepare_command(broadcast_table_query{query}, "broadcast_tables query");
     auto guard = group0_client.create_result_guard(group0_cmd.new_state_id);
-    co_await group0_client.add_entry_unguarded(std::move(group0_cmd));
+    co_await group0_client.add_entry_unguarded(std::move(group0_cmd), nullptr);
```
Comment (kbr-scylla) [inline service/broadcast_tables/experimental/lang.cc:33]: Please create a follow-up issue for passing abort_source here and inside raft_group0.cc where you passed `nullptr` to `modify_config`

**Review by kbr-scylla — APPROVED** [review]

**Comment by kbr-scylla** [comment]
Qd

File: service/broadcast_tables/experimental/lang.cc (hunk 1 above)
Comment (patjed41) [reply service/broadcast_tables/experimental/lang.cc:33]: Opened #16682

Comment (kbr-scylla) [reply service/broadcast_tables/experimental/lang.cc:33]: Thanks!

**Pull request merged**